tmt/steps/discover/shell.py

```diff
@@ -246,6 +246,16 @@ class DiscoverShell(tmt.steps.discover.DiscoverPlugin):
 
     _tests: List[tmt.base.Test] = []
 
+    @classmethod
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
+        """ Prepare command line options for given method """
+        return [
+            click.option(
+                '--sync-repo', default=False, is_flag=True,
+                help='Force the sync of the whole git repo. By default, the '
+                     'repo is copied only if the used options require it.'),
+            ] + super().options(how)
```

Comment:
Instead of adding the `options()` method I'd recommend to use the new `tmt.utils.field` to define the `DiscoverShellData` attribute `sync_repo` which will take care of the default and type as well. For inspiration have a look here:

https://github.com/teemtee/tmt/blob/902b641ca715d7f340c12c395506461b4f6e0ff5/tmt/steps/report/reportportal.py#L30-L36

Reply:
Related PR for `report` steps, BTW: https://github.com/teemtee/tmt/pull/1749/

Reply:
Thanks for reminding that one, @happz. Approved.

Reply:
@psss Couldn't we just merge this as is and convert options to field in a separate PR?

Reply:
Well, I'm not sure. Why adding it in the old way and converting it right away when the new way is easier, shorter and more clean? And it's just four line or so. Plus #1783 already outlines the other options.

Reply:
Attribute `sync_repo` is added via  the new `tmt.utils.field`.

Reply:
@idorax keep an eye on https://github.com/teemtee/tmt/pull/1784, it's related to how `field` handles values from CLI. If you start hacking on that follow-up patch, this might complicate your tests.